packages/uiweb/src/lib/components/space/SpaceWidget/SpaceWidget.tsx

```diff
@@ -1,11 +1,28 @@
-import React from 'react';
+import React, { useContext } from 'react';
+import styled from 'styled-components';
+
+import { ISpacesTheme } from '../theme';
+import { ThemeContext } from '../theme/ThemeProvider';
 
 export interface ISpaceWidgetProps {
   // Add props specific to the SpaceWidget component
-  temporary?: string; // just to remove eslint error of empty prop
+  theme?: ISpacesTheme;
```

Comment:
any reason for putting theme in props?

Reply:
Yes, kept the theme in ISpaceWidgetProps as the entire prop can be passed to the styled components, where the values can be used from the props. see line 25.

Alternatively, should I make a new interface just for this purpose?

Reply:
but, will this SpaceWidget component get theme in prop unless it's passed from SpaceUI class method?

Reply:
or, is it only for using type in styled components?
in this case, let's make a new interface for this

Reply:
yes it will get the prop irrespective of whether a custom theme is passed in the SpaceUI component or not, since the default prop is set to lightTheme, and providing a custom theme will only override it. So whether you pass it or not, the default theme will be available in context.

making a new interface for the same.